tests/unit/test-firefox/test.firefox.js

```diff
@@ -511,40 +512,240 @@ describe('firefox', () => {
+          () => Promise.resolve(false)
+        );
+        await firefox.useProfile('profileName', {
+          createProfileFinder,
+          isFirefoxDefaultProfile,
+        });
+        sinon.assert.calledOnce(isFirefoxDefaultProfile);
+        sinon.assert.calledWith(
+          isFirefoxDefaultProfile,
+          sinon.match('profileName')
+        );
       }
     ));
 
     it('configures a profile', () => withBaseProfile(
       (baseProfile) => {
         const configureThisProfile =
           sinon.spy((profile) => Promise.resolve(profile));
-        const app = 'fennec';
         const profilePath = baseProfile.path();
-        return firefox.useProfile(profilePath, {app, configureThisProfile})
+        return firefox.useProfile(profilePath, {configureThisProfile})
           .then((profile) => {
             sinon.assert.called(configureThisProfile);
             sinon.assert.calledWith(configureThisProfile, profile);
-            assert.equal(configureThisProfile.firstCall.args[1].app, app);
           });
       }
     ));
 
   });
 
+  describe('defaultCreateProfileFinder', () => {
+
+    function prepareReaderTest(readProfileReturns) {
+      const fakeReadProfiles = sinon.spy(() => {
+        return readProfileReturns;
+      });
+
+      const fakeGetPath = sinon.spy(() => Promise.resolve());
+      const fakeProfiles = [{Name: 'someName'}];
+      const userDirectoryPath = '/non/existent/path';
+
+      const FxProfile = {
+        Finder() {
+          return {
+            readProfiles: fakeReadProfiles,
+            getPath: fakeGetPath,
+            profiles: fakeProfiles,
+          };
+        },
+      };
+
+      return {
+        fakeReadProfiles,
+        fakeGetPath,
+        fakeProfiles,
+        FxProfile,
+        userDirectoryPath,
+      };
+    }
+
+    it('creates a finder', async () => {
+      const FxProfile = {
+        Finder: sinon.spy(function() {
+          return {};
+        }),
+      };
+      firefox.defaultCreateProfileFinder({FxProfile});
+      sinon.assert.calledWith(FxProfile.Finder, sinon.match(undefined));
+    });
+
+    it('creates finder based on userDirectoryPath if present', async () => {
+      const FxProfile = {
+        Finder: sinon.spy(function() {
+          return {};
+        }),
+      };
+      const userDirectoryPath = '/non/existent/path';
+      firefox.defaultCreateProfileFinder({userDirectoryPath, FxProfile});
+
+      sinon.assert.called(FxProfile.Finder);
+      sinon.assert.calledWith(
+        FxProfile.Finder,
+        sinon.match(userDirectoryPath),
+      );
+    });
+
+    it('returns a finder that resolves a profile name', async () => {
+      const {
+        fakeReadProfiles,
+        fakeGetPath,
+        FxProfile,
+        userDirectoryPath,
+      } = prepareReaderTest(Promise.resolve());
+
+      const getter = firefox.defaultCreateProfileFinder({
+        userDirectoryPath,
+        FxProfile,
+      });
+
+      await getter('someName');
```

Comment:
Just a minor tweak to this test:
we could test that the result value of getter('someName') is the same one that `fakeGetPath` returns.

Reply:
These changes have been applied in the follow up commit 7d8b703.